macros/REST_Track_PlotAlphaAna.C

```diff
@@ -0,0 +1,341 @@
+//
+//
+//
+#include <TRestRun.h>
+#include <TRestTrackAlphaAnalysisProcess.h>
+#include <TRestDetectorReadout.h>
+#include <TRestTrackEvent.h>
+#include <TRestAnalysisTree.h>
+#include <TRestDetectorHitsEvent.h>
+#include <TRestRawSignalEvent.h>
+
+#include "dirent.h"
+
```

Comment:
This method should not be implemented at the level of `tracklib`. If REST is not compiled with `detectorlib` or `rawlib` there will be problems when this macro is being loaded in `restRoot`.

Reply:
Indeed, you are rigth.

Do you prefer to move to general macros directory? I don't know what is the best strategy here.

Reply:
The libraries design is to focus on independent development of libraries. So that a particular library keeps a minimum of dependencies with other libraries. I am sure it is possible to think a way to keep that strategy alive.

In the case this is not possible, and using `rawlib` routines inside the `tracklib` is a must (although I would prefer to find a way to avoid) then the code instantiating `rawlib` must be encapsulated using a compilation flag, that could be created for example at `cmake/MacroRootDict.cmake` at `COMPILEDIR`, and define a `#ifdef RESTLIB_RAW`, not sure if this already exist. @nkx?

However, I would prefer that such a possibility does not exist, the present structure is the result of an effort to organise the code project, and I believe keeping isolated developments at the libraries as they are is the best to avoid future chaos. I understand this requires an effort, but I am open to discuss about possible solutions for the present implementation.

Keep present that if you add your macro to the `tracklib` pipeline, where only track routines are tested, the execution will fail.

https://gitlab.cern.ch/rest-for-physics/tracklib/-/pipelines/3637513

At the rawlib there are already a few tests running that do not need other libraries, we should push for something similar at tracklib

https://gitlab.cern.ch/rest-for-physics/tracklib/-/pipelines/3637513

Reply:
Adding it to the framework domain is also not desirable. The framework should know nothing about signal processing, physical track event reconstruction, detectors, etc. It should only implement basic analysis routines not connected to those domains. Such as fitting analysis tree observables, etc. This is discussed in the paper.

Reply:
To add dependency we can just call:
```
set(deps detector geant4 track raw)
COMPILELIB(deps)
```
Currently all libraries except connectorslib have no dependency on others. So it would be strange to make tracklib dependent on rawlib.